library/lcms2: build with libjpeg8-turbo

timmooney opened the pull request:
Update library/lcms2 to build with libjpeg8-turbo.

Beyond the standard changes (bump COMPONENT_REVISION, add CFLAGS/CXXFLAGS/LDFLAGS) I also ran 'gmake REQUIRED_PACKAGES' to add the missing dependencies.

The following executables & libraries within library/lcms2 now link to libjpeg8-turbo:

```bash
 /usr/bin/amd64/jpgicc
        libjpeg.so.8 =>  /usr/lib/libjpeg8-turbo/lib/amd64/libjpeg.so.8
/usr/bin/amd64/tificc
        libjpeg.so.8 =>  /usr/lib/libjpeg8-turbo/lib/amd64/libjpeg.so.8
/usr/bin/jpgicc
        libjpeg.so.8 =>  /usr/lib/libjpeg8-turbo/lib/libjpeg.so.8
/usr/bin/tificc
        libjpeg.so.8 =>  /usr/lib/libjpeg8-turbo/lib/libjpeg.so.8
```

Commit:
library/lcms2: build with libjpeg8-turbo

pyhalov commented on components/lcms2/Makefile:
```diff
@@ -60,3 +65,7 @@ BUILD_PKG_DEPENDENCIES = $(BUILD_TOOLS)
 
 include ../../make-rules/depend.mk
```
Style notes: Please, remove BUILD_PKG_DEPENDENCIES and include depend.mk. Also  move it to components/library/ and change  'include ../../make-rules/prep.mk' and similar to 'include $(WS_MAKE_RULES)/prep.mk

timmooney replied:
Just for my understanding: any component I find that is using either

`include ../../make-rules/prep.mk`

or

`include $(WS_TOP)/make-rules/prep.mk`

should be changed to instead use

`include $(WS_MAKE_RULES)/prep.mk`

(for configure.mk, ips.mk, etc. too).

pyhalov replied:
yes

timmooney commented:
In addition to the changes I already mentioned, the pull request now also includes:

- renamed components/lcms2 to components/library/lcms2 and updated the relative path to shared-macros.mk at the top of the Makefile.
- remove the include for depend.mk and the BUILD_PKG_DEPENDENCIES
- change all the includes for prep.mk, configure.mk, ips.mk, and lint-libraries.mk to use $(WS_MAKE_RULES)

Please review carefully.  This is the first time I've moved a component.
